[Jayden Kim] iP

Jayden-Kim-NUS opened the pull request:
### Arona :sob:
> It's time to get to work, Sensei! Arona has your back! - Arona ([Blue Archive](https://bluearchive.fandom.com/wiki/Arona))

Ever wish you had a self-proclaimed high-performance AI who helps you remember all the things you have to do?

Arona is
* text-based
* easy to learn
* ~~FAST~~ _**SUPER** FAST_ to use

All you need to do is
1. Download from here
2. Double click it
3. Add your tasks

And it is __FREE__!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice `Java` too. Here's the main method:
```java
public static void main(String[] args) {
    new Arona(".\\data.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Jayden-Kim-NUS committed:
Rename Duke to Arona

Jayden-Kim-NUS committed:
Edit greeting and exit text

Jayden-Kim-NUS committed:
Add echo and bye command

Jayden-Kim-NUS committed:
Add list and the "list" command

Jayden-Kim-NUS committed:
Add class Task, add mark and unmark command

Jayden-Kim-NUS committed:
add todos, events, deadlines

Jayden-Kim-NUS committed:
Add simple text ui testing

Jayden-Kim-NUS committed:
Add exception class and handling; add more tests

Jayden-Kim-NUS committed:
Add delete command, updated exceptions to use inbuilt methods

Jayden-Kim-NUS committed:
Add a data.txt file to save tasks

Jayden-Kim-NUS committed:
Add todo for future

Jayden-Kim-NUS committed:
Add todo to simplify code

Jayden-Kim-NUS committed:
Add comments for task file

Jayden-Kim-NUS committed:
Merge branch 'branch-Level-7'

Jayden-Kim-NUS committed:
Add date standards via LocalDate

Jayden-Kim-NUS committed:
Merge branch 'branch-Level-8'

Jayden-Kim-NUS committed:
Add Storage, TaskList, Parser, Ui class for OOP

Jayden-Kim-NUS committed:
Merge branch 'branch-A-MoreOOP'

Jayden-Kim-NUS committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yongkheehou commented on src/main/java/Arona/AronaException.java:
```diff
@@ -0,0 +1,6 @@
+package Arona;
```
How about creating a subpackage called `Arona.exceptions` to store all your exception-related classes in? You can maybe have subclasses that inherit from a general exception class and utilize some OOP there? 🤔

yongkheehou commented on src/main/java/Arona/Deadline.java:
```diff
@@ -0,0 +1,26 @@
+package Arona;
```
Yup, similar to this, perhaps all the Task-related classes should be in its own `Arona.tasks` package?

aldentantan replied:
Same, agree here about the usage of packages to organise and separate related classes

yongkheehou commented on src/main/java/Arona/Events.java:
```diff
@@ -0,0 +1,29 @@
+        return "[E]";
+    }
+
+    public String toFriendlyString() {
```
this line seems slightly long for our coding standard, perhaps it'll be good to split it into multiple lines instead?

yongkheehou commented on src/main/java/Arona/Parser.java:
```diff
@@ -0,0 +1,141 @@
+
+import java.time.LocalDate;
+
+public class Parser {
```
JavaDoc could be added to the public classes/ methods to improve documentation?

yongkheehou commented on src/main/java/Arona/Parser.java:
```diff
@@ -0,0 +1,141 @@
+        }
+
+        // Mark and unmark command
+        else if (input.toLowerCase().startsWith("mark") || input.toLowerCase().startsWith("unmark")) {
```
similarly here perhaps its better to split the part after the `||` into another line to make it cleaner?

yongkheehou reviewed (commented):
Overall the logic seems good but can try using CheckStyle to make some small improvements for code style

Jayden-Kim-NUS committed:
Add junit test files

Jayden-Kim-NUS committed:
Merge branch 'A-JUnit'

Jayden-Kim-NUS committed:
Add storage command

Jayden-Kim-NUS committed:
Add javadoc comments

Jayden-Kim-NUS committed:
Edit Storage class naming to follow standard

Jayden-Kim-NUS committed:
Add find command

Jayden-Kim-NUS committed:
Merge branch 'branch-A-JavaDoc'

Jayden-Kim-NUS committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Arona/Storage.java

Jayden-Kim-NUS committed:
Merge branch 'branch-Level-9'

Jayden-Kim-NUS committed:
Add more javadocs

aldentantan commented on src/main/java/Arona/Parser.java:
```diff
@@ -0,0 +1,162 @@
+        }
+
+        // Todos or events or deadline command
+        else if (input.toLowerCase().startsWith("todo ") || input.toLowerCase().startsWith("event ") || input.toLowerCase().startsWith("deadline ")) {
```
This line looks to be over 120 characters for the coding standard limit, would recommend breaking this into a new line

aldentantan commented on src/main/java/Arona/Arona.java:
```diff
@@ -0,0 +1,86 @@
+        }
+    }
+
+    /**
+     * The main loop of the chatbot
+     * <p>
+     * 1. Scans for user input
+     * <p>
+     * 2. Parses input and process them
+     * <p>
+     * 3. Catch exceptions
+     * <p>
+     * 4. Break loop when bye command is send
+     */
```
Clear and concise Javadocs to explain the run() loop

aldentantan reviewed (commented):
nice code it is very readable, the design is very human

Jayden-Kim-NUS committed:
Add gui, add enums, add new AronaExceptions

Jayden-Kim-NUS committed:
Merge branch 'branch-Level-10'

Jayden-Kim-NUS committed:
Fix build.gradle so jar files can be made

Jayden-Kim-NUS committed:
Add assertions in Arona and Storage class

Jayden-Kim-NUS committed:
Remove deep nesting in Parser class

Jayden-Kim-NUS committed:
Merge pull request #1 from Jayden-Kim-NUS/branch-A-Assertions

Add assertions in Arona and Storage class

Jayden-Kim-NUS committed:
Merge branch 'branch-A-CodeQuality'

Jayden-Kim-NUS committed:
Add BCD-Extension Archive command

Jayden-Kim-NUS committed:
Merge branch 'branch-BCD-Extension'

Jayden-Kim-NUS committed:
Add exception for archive command

MissingArgumentException class: case for "archive" command is missing

No exception is thrown when archive command has a blank name or a name
over 260 characters, file is not created and there is no error message.

Let's,
* Add new case to switch case inside MissingArgumentException
* Add detailed error message explaining what is wrong with input

Jayden-Kim-NUS committed:
Update save method in Storage class

save() method in Storage: rewrite to simplify

the save() method has largely similar code to archive() and uses the
clunky Files.write with Collections.singletonList that is hard to
understand and inefficient

Let's,
* combine save and archive into one method using varags to specify
which file name to save to
* use Files.writeString which can handle missing files and is easier
to understand as it accepts a String

Jayden-Kim-NUS committed:
Update README.md

Jayden-Kim-NUS committed:
Update README.md

Jayden-Kim-NUS committed:
Add more personality to responses

Jayden-Kim-NUS committed:
Merge remote-tracking branch 'origin/master'

Jayden-Kim-NUS committed:
Rename package to lowercase arona

Jayden-Kim-NUS committed:
Change app icon, fix test files

Jayden-Kim-NUS committed:
Update README.md

Jayden-Kim-NUS committed:
Update README.md

Jayden-Kim-NUS committed:
Update README.md
